### PR TITLE
DevContainer optimizations

### PR DESCRIPTION
**Pull request by guzman-raphael**
Supersedes #108 

- [x] Reduce image size (~4GB w/ `suite2p`, ~2GB w/o). Left `suite2p` in for now.
- [x] Improve build time through better caching
- [x] Persist MySQL data. To purge, users can do this manually or simply recreate the Codespace.
- [x] Add schema clean up function
- [x] Set notebook kernel
- [x] Cache docker images
- [x] Update environment launch instructions

### CODE REVIEW

**Review by kabilar — CHANGES_REQUESTED** [review]
Thanks @guzman-raphael.  One comment above.  I am currently testing on my fork.

**Comment by kabilar** [comment]
When I removed `pip install --no-cache-dir suite2p` from the Dockefile since we are installing suite2p from a forked version in the requirements.txt, there is a container error: `onCreateCommand failed with exit code 1`.

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+COPY ./requirements.txt /tmp/
+RUN \
     # workflow dependencies
     apt-get install gcc ffmpeg libsm6 libxext6 -y && \
-    # Commented out CaImAn installation due to Docker image size limits on Codespace
     # pip install --no-cache-dir Pygments==2.14.0  && \
     # git clone https://github.com/datajoint/CaImAn.git  && \
     # pip install --no-cache-dir -r ./CaImAn/requirements.txt  && \
     # pip install --no-cache-dir ./CaImAn && \
+    pip install --no-cache-dir suite2p && \
```
[the remark this answers is not in the document]
Comment (guzman-raphael) [reply .devcontainer/Dockerfile:35]: @kabilar I'd recommend us merging this for now and reviewing this again once we can discuss why Tolga's fork was needed and is so large. At least for the purposes of this demo, the included updates seem to work fine.

**Comment by guzman-raphael** [comment]
In light of the upcoming demo at Bonn University I'll merge as-is since it seems functional.

**Pull request merged**